File: officeenvironmentspyfrontend/BackEndServer.java
import java.io.*;
import java.net.*;

public class BackEndServer
{
    public static void main(String[] args) throws IOException, ClassNotFoundException
    {
	// Port number to bind server to.
	final int portNum = 8192;

	while(true)
	    {
		// Socket for server to listen at.
		ServerSocket listener = new ServerSocket(portNum);
		try
		    {
			// Accept a client connection once Server recives one.
			Socket clientSocket = listener.accept();

			// Input and output streams. Output stream must be created first.
			ObjectOutputStream out = new ObjectOutputStream(clientSocket.getOutputStream());
			ObjectInputStream in = new ObjectInputStream(clientSocket.getInputStream());

			// Read in and handle request (it should be a string.)
			String response = handleRequest((String)in.readObject());
			// Sending response back to client
			out.writeObject(response);

			out.close();
			in.close();
			clientSocket.close();
		    }
		finally
		    {		// Close server socket.
			listener.close();
		    }
	    }
    }

    private static String handleRequest(final String request)
    {
	System.out.println(request);
	switch(request)
	    {
	    case "get office stats":
		return new String("these are the stats");
	    default:
		return new String("Error: request not valid");
	    }
    }
}

File: officeenvironmentspyfrontend/BackEndServer.py
import socket
import urllib


def backendServer():
    HOST = socket.gethostname()
    PORT_NUM = 8192

    server_socket = socket.socket()
    server_socket.bind((HOST, PORT_NUM))  # bind host address and port together

    while True:
        # How many clients can the server listen to simultaneously?
        server_socket.listen(1)
        conn, address = server_socket.accept()  # Accept connection
        print("Connection from: " + str(address))
    
        while True:
            data = conn.recv(1024).decode()
            if not data:
                break
            print("Recived: request (" + data + ")") 
            if(data == "SITE STATS"):
                print("hey")
                opener = urllib.URLopener()
                print("hey")
                
                url = "https://s3.amazonaws.com/office-environment-spy-back-end-data/site_0.csv"
                print("hey")
                file = opener.open(url)
                print("hey")
                conn.send(file.read().replace('\n', '@SPECIAL@'))  # send data to the client
#                conn.send("Cats")  # send data to the client
                print("hey")
            else:
                data = "Error: recived unknown request (" + data + ")"
                conn.send(data.encode())

        conn.close()  # close the connection


if __name__ == '__main__':
    backendServer()

